akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -168,6 +230,12 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
 
         // both client and server must send a settings frame as first frame
         multiplexer.pushControlFrame(SettingsFrame(initialLocalSettings))
+
+        if (pingState.isDefined) {
+          // to limit overhead rather than constantly rescheduling a timer and looking at system time we use a constant timer
+          // FIXME does that really matter (I think the idle timeout logic touches nanotime for every frame)?
+          schedulePeriodically(ConfigurablePing.Tick, 1.second)
```

Comment:
What would happen if we just take `pingInterval` as the tick interval to allow shorter intervals? We could document that `pingTimeout` will be checked in those same units (or even use the minimum of `pingInterval` and `pingTimeout`).

Reply:
Yeah maybe that's better, and more efficient for actual intervals you'd want which may be more like 1 per minute or so.